src/JetsProducer.cc

```diff
@@ -1,21 +1,93 @@
+        vtx3DSig.push_back(jet.userFloat("vtx3DSig"));
+        float vtxPx = jet.userFloat("vtxPx");
+        float vtxPy = jet.userFloat("vtxPy");
+        float vtxPt_ = std::sqrt(std::pow(vtxPx,2) + std::pow(vtxPy,2));
+        vtxPt.push_back(vtxPt_);
+        vtxNtracks.push_back(jet.userFloat("vtxNtracks"));
+        // some loops needed to dig some information within the jet itself
+        std::vector<std::pair<float, int>> softLeptInJet; // pt, index
+        softLeptInJet.clear();
+        float leadTrackPt_ = 0.;
+        for (unsigned int idaughter = 0; idaughter < jet.numberOfDaughters(); idaughter++)
+        {
+            const reco::Candidate * daughter = jet.daughter(idaughter);
+            if (std::abs(daughter->pdgId()) == 11 || std::abs(daughter->pdgId()) == 13)
+                softLeptInJet.push_back( std::make_pair(daughter->pt(), idaughter) );
+            if (daughter->charge() != 0)
+            {
+                float leadTrackPt_tmp = daughter->pt();
+                if(leadTrackPt_tmp > leadTrackPt_)
+                    leadTrackPt_ = leadTrackPt_tmp;
+            }
+        }
+        leadTrackPt.push_back(leadTrackPt_);
+        float leptonPtRel_ = -1.;
+        float leptonPt_ = -1.;
+        float leptonDeltaR_ = -1.;
+        int softLeptIdx_ = -1;
+        if (softLeptInJet.size() > 0)
+        {
+            std::sort(softLeptInJet.begin(), softLeptInJet.end());
+            softLeptIdx_ = softLeptInJet.back().second;
+            const reco::Candidate *daughter = jet.daughter(softLeptIdx_);
+            leptonPt_ = daughter->pt();
+            leptonPtRel_ = daughter->pt() / jet.pt();
+            leptonDeltaR_ = deltaR(*daughter, jet);
+        }
+        leptonPtRel.push_back(leptonPtRel_);
+        leptonPt.push_back(leptonPt_);
+        leptonDeltaR.push_back(leptonDeltaR_);
+        chargedMultiplicity.push_back(jet.chargedMultiplicity());
+        // The regression output itself
+        if (computeRegression)
+        {
+            Jet_pt = jet.pt();
+            Jet_corr = jet.jecFactor("Uncorrected");
+            rho = *rho_handle;
+            Jet_eta = jet.eta();
+            Jet_mt = jet.mt();
+            Jet_leadTrackPt = leadTrackPt_;
+            Jet_leptonPtRel = leptonPtRel_;
+            Jet_leptonPt = leptonPt_;
+            Jet_leptonDeltaR = leptonDeltaR_;
+            Jet_neHEF = jet.neutralHadronEnergyFraction();
+            Jet_neEmEF = jet.neutralEmEnergyFraction();
+            Jet_chMult = jet.chargedMultiplicity();
+            Jet_vtxPt = vtxPt_;
+            Jet_vtxMass = jet.userFloat("vtxMass");
+            Jet_vtx3dL = jet.userFloat("vtx3DVal");
+            Jet_vtxNtrk = jet.userFloat("vtxNtracks");
+            Jet_vtx3deL = jet.userFloat("vtx3DSig");
+            regPt.push_back((bjetRegressionReader->EvaluateRegression("BDTG method"))[0]);
+        } else {
+            regPt.push_back(jet.pt());
+        }
+
+        if (! pass_cut(jet))
```

Comment:
Unfortunately it won't work, because as soon as a jet does not pass the cut, arrays will be desynchronized (arrays before the cut will have one entry more than those after).

It's tricky to fix, but the easiest in this case would be to move this check... before the regression. It's ok in this case since we don't rescale the p4 anymore, so after / before is the same, but before fixes the desynchronization issue.